packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_TotalElementLoop_DP.cpp

```diff
@@ -306,17 +304,37 @@ int main (int argc, char *argv[])
   //Tpetra::MatrixMarket::Writer<matrix_t>::writeSparse(ofs, crs_matrix);
   //ofs.close();
 
+  return 0;
+}
+
+
+
+int main (int argc, char *argv[]) 
+{
+  int status = EXIT_SUCCESS;
+
+  // MPI boilerplate
+  Tpetra::initialize(&argc, &argv);
+  COMM_T comm = Tpetra::DefaultPlatform::getDefaultPlatform ().getComm();
+
+  // Entry point
+  if(execute(comm))
+  {
+    status = EXIT_FAILURE;
+  }
+
   // Print out timing results.
-  TimeMonitor::report(comm.ptr(), std::cout, "");
+  Teuchos::TimeMonitor::report(comm.ptr(), std::cout, "");
 
   // Finalize
   Tpetra::finalize();
 
   // This tells the Trilinos test framework that the test passed.
   if(0 == comm->getRank())
```

Comment:
You can't call `Comm` methods after MPI has been finalized.  `Tpetra::finalize()` may finalize MPI (it does if and only if `Tpetra::initialize(...)` initialized MPI).